gitlab.go

```diff
@@ -720,12 +729,23 @@ type Response struct {
 	CurrentPage  int
 	NextPage     int
 	PreviousPage int
+
+	PreviousLink string
+	NextLink     string
+	FirstLink    string
+	LastLink     string
+
+	// The "next" link in the Link header includes query parameters for the
+	// next paginated request, but the relevant parameters vary among API
+	// endpoints.
+	NextLinkParameters url.Values
```

Comment:
Minor question: I wonder if this field should be omitted, so that `BuildKeysetPaginatedRequestOptionFunc` can just take in the `NextLink` and lazily parse it.

Reply:
I think that's a good idea. It seems like the goal of the library right now is more "provide a clean interface without being too opinionated" rather than "provide as little as possible" or "all batteries included." I'll see if I can streamline that a little better.